Influenxio/docker-airflow
# add mysql env for docker compose to test

# How to test

1. add `.env` and put your mysql account in that file to get connect to influenxio db
```
USERNAME=xxxxx
PASSWORD=xxxx
```
2. run docker compose to start the airflow service but note that it will use postgres db as the airflow db so the dag and task status would be different from production airflow db (which also a good separation from production db)
```
docker-compose -f docker-compose-LocalExecutor.yml up
```

3a. do any of your code change in `dags` folder and it will automatically change in airflow UI so that you can test
3b. do any of your code change in `dags` folder and you can connect into the docker to test
```
docker ps  # get the docker container id
docker exec -it {container_id} bash
# below are some example of airflow test commands
airflow list_dags
airflow render {dag_id} {task_id} {exec_ds}
airflow test {dag_id} {task_id} {exec_ds}
```

# Reviewer
@kstseng
